### PR TITLE
chore: Add missing javadocs to Schedule Service

### LINKED ISSUE #13601: Add missing Javadocs in Schedule service

### PR DESCRIPTION
**Pull request by thomas-swirlds-labs**
**Description**:

Adds javadocs to the ScheduleService

**Related issue(s)**:

Fixes #13601

### CODE REVIEW

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/codec/ScheduleServiceStateTranslator.java [hunk 1]
```diff
@@ -154,6 +172,12 @@ public static com.hedera.node.app.service.mono.state.virtual.schedule.ScheduleVi
         return pbjToState(optionalSchedule);
     }
 
+    /**
+     * Migrates the state of the schedule service from the scheduleId to the schedule virtual value
```
Comment (mhess-swl) [inline hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/codec/ScheduleServiceStateTranslator.java:176]:
```suggestion
     * Converts a {@link Schedule} object to a {@link ScheduleVirtualValue}
```

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java [hunk 2]
```diff
@@ -60,12 +60,21 @@ abstract class AbstractScheduleHandler {
...
      */
     protected static record ScheduleKeysResult(Set<Key> updatedSignatories, Set<Key> remainingRequiredKeys) {}
 
+    /**
+     * Gets the set of all the keys required to sign a transaction.
+     *
+     * @param scheduleInState the schedule in state
+     * @param context    the Prehandle context
+     * @return the set of keys required to sign the transaction
+     * @throws PreCheckException the pre check exception
```
Comment (mhess-swl) [inline hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java:76]: I'd prefer a little more for the `@throws`, or to not have it at all. Seems like it's just clutter the way it's written

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java [hunk 3]
```diff
@@ -78,6 +87,14 @@ protected Set<Key> allKeysForTransaction(
         return getKeySetFromTransactionKeys(keyStructure);
     }
 
+    /**
+     * Get the schedule keys result to sign the transaction
+     *
+     * @param scheduleInState the schedule in state
+     * @param context         the Prehandle context
+     * @return the schedule keys result containing the updated signatories and the remaining required keys
+     * @throws HandleException the handle exception
```
Comment (mhess-swl) [inline hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java:96]: Same comment as `@throws PreCheckException` above. There seem to be a few of these.

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleHandlers.java [hunk 4]
```diff
@@ -33,6 +36,14 @@ public class ScheduleHandlers {
 
     private final ScheduleSignHandler scheduleSignHandler;
 
+    /**
+     * Instantiates a new Schedule handlers.
```
Comment (mhess-swl) [inline hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/ScheduleHandlers.java:40]:
```suggestion
     * Instantiates a new Schedule handler.
```

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/schemas/V0490ScheduleSchema.java [hunk 5]
```diff
@@ -64,6 +64,9 @@ public final class V0490ScheduleSchema extends Schema {
      */
     private static MerkleScheduledTransactions fs;
 
+    /**
+     * Instantiates a new V0490 schedule schema. The version is set to 0.49.0.
```
Comment (mhess-swl) [inline hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/schemas/V0490ScheduleSchema.java:68]:
```suggestion
     * Instantiates a new V0490 (version 0.49.0) schedule schema.
```

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/schemas/V0490ScheduleSchema.java [hunk 6]
```diff
@@ -181,6 +184,11 @@ private static StateDefinition<ProtoBytes, ScheduleList> schedulesByEquality() {
                 SCHEDULES_BY_EQUALITY_KEY, ProtoBytes.PROTOBUF, ScheduleList.PROTOBUF, MAX_SCHEDULES_BY_EQUALITY);
     }
 
+    /**
+     * Used to migrate the state to the new schema. It is not thread safe and is set to null after migration.
+     *
+     * @param fs the fs
```
Comment (mhess-swl) [inline hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/schemas/V0490ScheduleSchema.java:190]: FYI `fs` is short for `from state` and `ts` is short for `to state`

File: hedera-node/hedera-schedule-service/src/main/java/com/hedera/node/app/service/schedule/ScheduleServiceDefinition.java [hunk 7]
```diff
@@ -33,8 +33,14 @@
  */
 @SuppressWarnings("java:S6548")
 public final class ScheduleServiceDefinition implements RpcServiceDefinition {
+    /**
+     * The constant ScheduleServiceDefinition INSTANCE.
```
Comment (mhess-swl) [inline hedera-node/hedera-schedule-service/src/main/java/com/hedera/node/app/service/schedule/ScheduleServiceDefinition.java:37]:
```suggestion
     * The global singleton ScheduleServiceDefinition INSTANCE.
```

File: hedera-node/hedera-schedule-service/src/main/java/com/hedera/node/app/service/schedule/WritableScheduleStore.java [hunk 8]
```diff
@@ -22,27 +22,43 @@
...
+     * Delete a given schedule from this state. Given the ID of a schedule and a consensus time, delete that ID from
+     * this state as of the consensus time {@link Instant} provided.
+     *
      * @param scheduleToDelete The ID of a schedule to be deleted.
-     * @param consensusTime The current consensus time
-     * @throws IllegalStateException if the {@link ScheduleID} to be deleted is not present in this state,
-     *     or the ID value has a mismatched realm or shard for this node.
+     * @param consensusTime    The current consensus time
+     * @return the schedule
+     * @throws IllegalStateException if the {@link ScheduleID} to be deleted is not present in this state,     or the ID
```
Comment (mhess-swl) [inline hedera-node/hedera-schedule-service/src/main/java/com/hedera/node/app/service/schedule/WritableScheduleStore.java:37]:
```suggestion
     * @throws IllegalStateException if the {@link ScheduleID} to be deleted is not present in this state,  or the ID
```
Nit

**Review by mhess-swl — COMMENTED** [review]
Just a few nits throughout. @thomas-swirlds-labs I'll leave it up to you which ones you think should be addressed (if any)

**Review by jsync-swirlds — COMMENTED** [review]
My apologies if I misunderstand the purpose of adding all this new javadoc.  I'm assuming, here, that the goal is to have useful and complete javadoc, not just to have "something" to meet a build requirement.

To that end, I've noted some general recommendations for improving the value of this effort, including some suggested approaches to writing more useful and complete API specification (which is what Javadoc is supposed to be).

I did also note a couple items that can be removed, either now or in the near future.

Oracle's [javadoc guidelines](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html) can be very helpful in writing thorough and effective javadoc.

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java (hunk 2 above)
Comment (thomas-swirlds-labs) [reply hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java:76]: Thanks @mhess-swl. Updated.

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/ScheduleStoreUtility.java [hunk 9]
```diff
@@ -31,9 +31,20 @@
...
 
+    /**
+     * Calculate bytes hash of a schedule based on the schedule's memo, admin key, scheduled transaction, expiration
+     * time, and wait for expiry flag.
+     *
+     * @param scheduleToHash the schedule to hash
+     * @return the bytes
+     */
     // @todo('7773') This requires rebuilding the equality virtual map on migration,
     //      because it's different from ScheduleVirtualValue (and must be, due to PBJ shift)
```
[the remark this answers is not in the document]
Comment (thomas-swirlds-labs) [reply hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/ScheduleStoreUtility.java:49]: Removed and issue closed. Thanks @jsync-swirlds !

File: hedera-node/hedera-schedule-service/src/main/java/com/hedera/node/app/service/schedule/WritableScheduleStore.java [hunk 10]
```diff
@@ -22,27 +22,43 @@
 import edu.umd.cs.findbugs.annotations.Nullable;
 import java.time.Instant;
 
+/**
+ * The interface Writable schedule store.
+ */
 public interface WritableScheduleStore extends ReadableScheduleStore {
 
     /**
-     * Delete a given schedule from this state.
```
[the remark this answers is not in the document]
Comment (thomas-swirlds-labs) [reply hedera-node/hedera-schedule-service/src/main/java/com/hedera/node/app/service/schedule/WritableScheduleStore.java:28]: thanks! I have re-instated the newline now 👍

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java [hunk 11]
```diff
@@ -60,12 +60,21 @@ abstract class AbstractScheduleHandler {
 
     /**
      * A simple record to return both "deemed valid" signatories and remaining primitive keys that must sign.
-     * @param updatedSignatories a Set of "deemed valid" signatories, possibly updated with new entries
+     *
+     * @param updatedSignatories    a Set of "deemed valid" signatories, possibly updated with new entries
      * @param remainingRequiredKeys A Set of Key entries that have not yet signed the scheduled transaction, but
-     *     must sign that transaction before it can be executed.
+     *                              must sign that transaction before it can be executed.
```
[the remark this answers is not in the document]
Comment (thomas-swirlds-labs) [reply hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/handlers/AbstractScheduleHandler.java:66]: 💯

File: hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/codec/ScheduleServiceStateTranslator.java [hunk 12]
```diff
@@ -42,11 +42,21 @@
 import java.util.Objects;
 import java.util.Optional;
 
+/**
+ * Utility class used for conversion of schedule virtual values to schedules.
+ */
 public final class ScheduleServiceStateTranslator {
```
[the remark this answers is not in the document]
Comment (thomas-swirlds-labs) [reply hedera-node/hedera-schedule-service-impl/src/main/java/com/hedera/node/app/service/schedule/impl/codec/ScheduleServiceStateTranslator.java:48]: Have set it to deprecated. Thanks @jsync-swirlds 👍

**Review by mhess-swl — APPROVED** [review]

**Pull request merged**